app/gui2/src/util/ast/abstract.ts

```diff
@@ -885,6 +885,25 @@ export class Function extends Ast {
     this.equals_ = equals
     this.body_ = body
   }
+
+  static new(
+    module: MutableModule,
+    name: string,
+    args: Ast[],
+    exprs: Ast[],
+    trailingNewline?: boolean,
+  ): Function {
+    const exprs_: BlockLine[] = exprs.map((expr) => ({ expression: { node: expr } }))
+    if (trailingNewline) {
+      exprs_.push({ newline: { node: Token.new('\n') }, expression: null })
+    }
+    const body = BodyBlock.new(exprs_, module)
+    const args_ = args.map((arg) => [{ node: arg.exprId }])
```

Comment:
When constructing an Ast object it's important to ensure any children are accessible in the object's module. After #8711 the `makeChild` helper takes care of this; it can be used like: `node: makeChild(module, arg, id)`. `id` should be the ID of the object currently being constructed; it can be generated with `newAstId()` and used here and as the ID argument to the `Function` constructor call below.